packages/components/src/components/stepper-horizontal/stepper-horizontal-item/stepper-horizontal-item-styles.ts

```diff
@@ -74,96 +71,99 @@ export const getComponentCss = (state: StepperState, disabled: boolean, theme: T
-                borderRadius: '50%',
-                ...(isDisabled
-                  ? {
-                      boxSizing: 'border-box',
-                      border: `1px solid ${disabledColor}`,
-                    }
-                  : {
-                      background: primaryColor,
-                    }),
-              },
-              '&::after': {
-                // value of counter element
-                content: '""',
-                position: 'absolute',
-                top: '50%',
-                transform: 'translateY(-50%)',
-                left: counterValuePosition,
-                width: counterValueSize,
-                height: counterValueSize,
-                background: `${getSvg(
-                  isDisabled ? disabledColor : invertedBaseColor
-                )} 0 50% / ${spriteWidth} ${spriteHeight} no-repeat`,
-              },
-            }
-          : // other icons via icon component
-          isDisabled
-          ? {
-              cursor: 'not-allowed',
-              textDecoration: 'none',
-            }
-          : {
-              cursor: 'pointer',
-              textDecoration: 'underline',
-              ...hoverMediaQuery({
-                ...getHoverJssStyle(),
-                '&:hover .icon': {
-                  color: hoverColor,
-                },
+        width: 'max-content',
+        cursor: isDisabled ? 'not-allowed' : 'pointer',
+        // mergeDeep needed because of hoverMediaQuery in certain modes not wrapping keys and therefore overriding "&::before" key
+        ...mergeDeep(
+          {
+            '&::before': {
+              content: '""',
+              position: 'absolute',
+              ...getInsetJssStyle(0),
+              borderRadius: borderRadiusSmall,
+              zIndex: '-1',
+              ...(isStateCurrent && {
+                ...frostedGlassStyle,
+                background: hoverColor,
               }),
-            }),
+            },
+          },
+          !isStateCurrentOrUndefined && !isDisabled
+            ? hoverMediaQuery({
+                '&::before': {
+                  transition: getTransition('background'),
+                },
+                '&:hover::before': {
+                  ...frostedGlassStyle,
+                  background: hoverColor,
+                },
+              })
+            : {
+                // counter
+                '&::after': {
+                  gridArea: '1 / 1 / 1 / 1',
+                  content: '""',
+                  position: 'absolute',
+                  height: fontLineHeight,
+                  width: fontLineHeight,
+                },
+              }
+        ),
+        '&:focus::before': {
+          border: `${borderWidthBase} solid ${focusColor}`,
+          borderRadius: borderRadiusSmall,
+        },
+        '&:focus:not(:focus-visible)::before': {
+          borderColor: 'transparent',
+        },
       },
     },
+    label: {
```

Comment:
With the suggested approach or even a `display: flex` there wouldn't be any need for an additional selector and additional node.